libvast/include/vast/plugin.hpp

```diff
@@ -112,8 +112,12 @@ class plugin {
 
   /// Initializes a plugin with its respective entries from the YAML config
   /// file, i.e., `plugin.<NAME>`.
-  /// @param config The relevant subsection of the configuration.
-  [[nodiscard]] virtual caf::error initialize(data config) = 0;
+  /// @param plugin_config The relevant subsection of the configuration.
+  /// @param global_config The entire VAST configuration for potential access to
+  /// global options.
+  [[nodiscard]] virtual caf::error
+  initialize(data plugin_config, data global_config)
```

Comment:
Does it make sense to pass both as a const-ref so we don't copy a large data object for every single plugin initialization, and the plugin subsection twice at that?

Reply:
Additionally, should they both be `const record&` rather than `data`? At least for the global config, we always already have them in a record, so stuffing that back into a data just causes plugins to needlessly check whether the data contains a record. The same goes for the plugin-specific config, where if not present we can pass in an empty record instead, avoiding the need to check in each plugin.

Reply:
Yes, they should; I was initially doubting whether the lifetime of the `record`s would suffice for plugin initialization, but yes, they would because each plugin processes the original `record` by converting it into a `configuration` anyway. I changed the signature in https://github.com/tenzir/vast/pull/2949/commits/b51d8b9b88beab81ae3cf5be88ec8a55195f6d71